proxy/http/HttpSM.cc

```diff
@@ -5460,9 +5460,14 @@ HttpSM::handle_http_server_open()
   //          server session's first transaction.
   if (nullptr != server_session) {
     NetVConnection *vc = server_session->get_netvc();
-    if (vc != nullptr && (vc->options.sockopt_flags != t_state.txn_conf->sock_option_flag_out ||
-                          vc->options.packet_mark != t_state.txn_conf->sock_packet_mark_out ||
-                          vc->options.packet_tos != t_state.txn_conf->sock_packet_tos_out)) {
+
+    SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(vc);
+    if (ssl_vc) {
+      ssl_vc->setClientVerifyEnable(t_state.txn_conf->ssl_client_verify_server);
+    }
+    if (vc != NULL && (vc->options.sockopt_flags != t_state.txn_conf->sock_option_flag_out ||
```

Comment:
I agree with the point that we should not move HttpSM configs to other subsystems. But with the increasing amount of configuration requirements (specially ssl) based on origin server, it would be better if we have a separate config for origin servers only. That way, ATS can populate a global storage just like other configs and all the subsystems can access those . Of course, with this comes the issues of memory, computation, reloading and all those things